posthog/migrations/0893_backfill_partition_format.py

```diff
@@ -0,0 +1,40 @@
+# generated manually to backfill partition formats to "month" for existing syncs
+# in the data warehouse product
+
+from django.db import migrations
+from django.db.models import Q
+
+from posthog.warehouse.models import ExternalDataSchema as ExternalDataSchemaModel
+
+
+def forwards(apps, _):
+    ExternalDataSchema: ExternalDataSchemaModel = apps.get_model("posthog", "ExternalDataSchema")
+    affected_schemata = ExternalDataSchema.objects.filter(
+        Q(
+            deleted=False,
+            sync_type_config__partitioning_enabled=True,
+            sync_type_config__partition_mode="datetime",
+            sync_type_config__partition_format__isnull=True,
+        )
+    )
+    for schema in affected_schemata:
+        schema.sync_type_config["partition_format"] = "month"
+        schema.sync_type_config["backfilled_partition_format"] = True
```

Comment:
If you'd like to use this, which I think is a good idea, then i'd suggest also adding this key to our `sync_type_config` reset method here: https://github.com/PostHog/posthog/blob/master/posthog/warehouse/models/external_data_schema.py#L201-L213. That way, we clear the `backfilled_partition_format` key if a schema gets fully reset (which is what we want)